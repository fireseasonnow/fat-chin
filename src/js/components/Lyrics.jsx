// @flow
import React from 'react';
import styled from 'styled-components'

type Props = {
    name: string,
    text: string,
    hideLyricsHandler: (e: Object) => void
};

const lyrics = (props: Props) => (
    <Inner>
        <Close onClick={(e) => props.hideLyricsHandler(e)} type="button"></Close>
        <Text><Heading>{props.name}</Heading> {props.text ? props.text : 'Instrumental'}</Text>
    </Inner>
);

export default lyrics;

const Inner = styled.div`
    display: grid;
    grid-template-columns: 1fr auto 1fr;
    height: 100%;
    overflow-y: scroll;
`;

const Close = styled.button`
    align-self: start;
    top: 42px;
    align-self: center;
    justify-self: end;
    background: none;
    color: #fff;
    border: 0;
    position: relative;
    left: -15px;
    width: 40px;
    height: 40px;
    outline: 0;
    padding-right: 28px;

    &:hover {
        cursor: pointer;
    }

    &:before {
        content: '';
        position: absolute;
        top: 20px;
        width: 21px;
        height: 1px;
        background-color: currentColor;
        transform: rotate(-45deg);
    }

    &:after {
        content: '';
        position: absolute;
        top: 20px;
        width: 21px;
        height: 1px;
        background-color: currentColor;
        transform: rotate(45deg);
    }
`;

const Text = styled.p`
    margin: 50px 0;
    white-space: pre-line;
`;

const Heading = styled.span`
    display: block;
    text-transform: uppercase;
    margin-bottom: 20px;
`;
