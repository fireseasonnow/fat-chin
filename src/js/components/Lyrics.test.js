import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';

import Lyrics from './Lyrics';

configure({ adapter: new Adapter() });

describe('<Lyrics />', () => {
    let wrapper = {};

    beforeEach(() => {
        wrapper = shallow(<Lyrics />);
    });

    test('should render without crashing', () => {
        expect(wrapper).toBeTruthy();
    });

    test('should render lyrics with name and text', () => {
        wrapper.setProps({
            name: 'Hello',
            text: 'Lorem Ipsum',
        });

        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    test('when lyrics are missing should render lyrics with name and "Instrumental" text', () => {
        wrapper.setProps({
            name: 'Hello',
            text: '',
        });

        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    test('when lyrics are missing should render lyrics with name and "Instrumental" text', () => {
        wrapper.setProps({
            name: 'Hello',
            text: '',
        });

        const check = wrapper.find('.qa-lyrics__text').contains('Instrumental');

        expect(check).toBeTruthy();
    });

    test('should fire hideLyricsHandler when clicked on a button', () => {
        const hideLyricsHandler = jest.fn();

        wrapper.setProps({
            hideLyricsHandler,
        });

        const event = { target: { value: 'test' } };
        wrapper.find('.qa-lyrics__close-btn').simulate('click', event.target.value);

        expect(hideLyricsHandler).toHaveBeenCalledWith(event.target.value);
    });
});
