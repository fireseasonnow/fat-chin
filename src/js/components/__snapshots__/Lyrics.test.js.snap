// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Lyrics /> should render lyrics with name and text 1`] = `
<div
  className="cover-zoom__inner lyrics__inner"
>
  <button
    className="cover-zoom__close lyrics__close-btn qa-lyrics__close-btn"
    onClick={[Function]}
    type="button"
  />
  <p
    className="lyrics__text qa-lyrics__text"
  >
    <span
      className="lyrics__heading"
    >
      Hello
    </span>
     
    Lorem Ipsum
  </p>
</div>
`;

exports[`<Lyrics /> when lyrics are missing should render lyrics with name and "Instrumental" text 1`] = `
<div
  className="cover-zoom__inner lyrics__inner"
>
  <button
    className="cover-zoom__close lyrics__close-btn qa-lyrics__close-btn"
    onClick={[Function]}
    type="button"
  />
  <p
    className="lyrics__text qa-lyrics__text"
  >
    <span
      className="lyrics__heading"
    >
      Hello
    </span>
     
    Instrumental
  </p>
</div>
`;
